frontend-react/src/components/Table/TableFilters.tsx

```diff
@@ -25,13 +25,14 @@ interface SubmissionFilterProps {
     onFilterClick?: ({ from, to }: { from: string; to: string }) => void;
 }
 
-/* This helper ensures start range values are inclusive
- * of the day set in the date picker. */
-const inclusiveDateString = (originalDateStr: string) => {
-    return new Date(
-        new Date(originalDateStr).setUTCHours(23, 59, 59, 999)
-    ).toISOString();
-};
+const DATE_RE = /^[0-9]{1,2}\/[0-9]{1,2}\/[0-9]{4}$/;
+
```

Comment:
Does a valid year HAVE to have the full year associated with it? ie, `01/01/2023` vs `01/01/23`?

Also, since we're checking with `Date.parse()` which checks for valid dates (no, months that don't exist, proper formatting etc), what is `DATE_RE` offering on top? Just curious! Might be worth a comment since I'm missing it.

Reply:
We should probably favor using momentjs to validate the dates rather than regex. see: https://momentjs.com/docs/#/parsing/string-format/

Reply:
Correct me if i'm wrong @stephenkao but we wanted to slowly move away from Moment JS since it's unmaintained now? IE [from their docs](https://momentjs.com/docs/#/-project-status/):

```
We now generally consider Moment to be a legacy project in maintenance mode. It is not dead, but it is indeed done.
```

Reply:
Interesting. I was not aware of its transition to maintenance mode. If there is a way to handle validation cleanly with the default Date object then that's def OK, I just assumed it probably wasn't. If it's not, we may need to use momentjs even if temporary if we decide to move to another datetime library instead of waiting for temporal.

Reply:
> Does a valid year HAVE to have the full year associated with it? ie, `01/01/2023` vs `01/01/23`?
> 
> Also, since we're checking with `Date.parse()` which checks for valid dates (no, months that don't exist, proper formatting etc), what is `DATE_RE` offering on top? Just curious! Might be worth a comment since I'm missing it.

@etanb No problem!  `Date` has different implementations across browsers, and the regular expression is just an extra check to make sure the user is inputting something that can be understood by all of them.  Like `new Date('11')` returns a Date instance of 11/01/2001 in Chrome but throws an `Invalid Date` error in Firefox.  It's not absolutely necessary, but it's just an added check so I can definitely add a comment explaining that + update to allow for two or four year digits.

@jpandersen87 Yeah, I think we should be at least reevaluating moment.js because it doesn't provide much more functionality than what the browser API offers now -- at least not for our use cases.  I'm personally okay with using a regular expression in lieu of moment.js for this particular instance since it's checking against the format of the input rather than the date itself, which is handled through `Date.parse`.  

We still have references to moment.js in our codebase so I'm not advocating for its immediate removal; I just want us to keep in mind that it may not be necessary, and we can reevaluate its inclusion in this project in a few months once we have a better understanding of how much we really need it.  What are your thoughts?

Reply:
Updated [here](https://github.com/CDCgov/prime-reportstream/pull/8348/commits/1ad2e1d2867e9bd9102f1c4f317d8be449a9603d).  Let me know what y'all think!